thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/DataResource.java

```diff
@@ -543,6 +547,81 @@ public WowSummary getWowSummary(
     return wowSummary;
   }
 
+  @GET
+  @Path("anomalies/ranges")
+  public Map<Long, List<TimeRange>> getAnomalyTimeRangesByMetricIds(
+      @QueryParam("metricIds") String metricIds,
+      @QueryParam("start") Long start,
+      @QueryParam("end") Long end) {
+
+    if (metricIds == null)
+      throw new IllegalArgumentException("Must provide metricIds");
+
+    if (start == null)
+      throw new IllegalArgumentException("Must provide start timestamp");
+
+    if (end == null)
+      throw new IllegalArgumentException("Must provide end timestamp");
+
+    List<Long> ids = new ArrayList<>();
+    for (String metricId : metricIds.split(",")) {
+      ids.add(Long.parseLong(metricId));
+    }
+
+    Map<Long, List<MergedAnomalyResultDTO>> anomalies = DAO_REGISTRY.getMergedAnomalyResultDAO().findAnomaliesByMetricIdsAndTimeRange(ids, start, end);
+
+    Map<Long, List<TimeRange>> output = new HashMap<>();
+    for(Map.Entry<Long, List<MergedAnomalyResultDTO>> entry : anomalies.entrySet()) {
+      output.put(entry.getKey(), truncateRanges(extractAnomalyTimeRanges(entry.getValue()), start, end));
+    }
+
+    return output;
+  }
+
+  static List<TimeRange> extractAnomalyTimeRanges(List<MergedAnomalyResultDTO> anomalies) {
```

Comment:
javadocs for these two (extractAnomalyTimeRanges  and truncateRanges) would be helpful